monorepo/monorepo-ws/apps/federation-gateway/src/services/server-manager.ts

```diff
@@ -1,52 +1,48 @@
 import { ApolloGateway } from "@apollo/gateway";
 import { ApolloServer } from "apollo-server";
-import { checkActualSuperGraph } from ".";
 import { interval, port } from "../configuration/config";
-import { SupergraphStorageManager } from "./interfaces";
 import { tryCreatePartOfSupergraph } from "./supergraph-services";
+import { SupergraphManager } from "@monorepo-ws/supergraph-manager"
+import { logger } from "libs/server/src/logger";
 
 
 export class ServerManager {
-    superGraphManager: SupergraphStorageManager
-    superGraphSchema: Buffer
     gateway: ApolloGateway
     server: ApolloServer
 
-    constructor(superGraphManager: SupergraphStorageManager){
-        this.superGraphManager = superGraphManager
-    }
-
-    private async checkSupergraph(){
-        const newFile = await this.superGraphManager.getSupergraph()
-        if(!checkActualSuperGraph(this.superGraphSchema, newFile)){
-            await this.restartServer(newFile)
-        }
+    constructor(private readonly supergraphManager: SupergraphManager){
     }
 
     async runApp() {
         await this.runServer()
-        setInterval(() => this.checkSupergraph, interval)
+        setInterval(this.runServer.bind(this), interval)
     }
 
     private async runServer() {
+        if(!this.server){
+            try {
```

Comment:
❗️takovej pro tip: u try catch delej vzdy to ze try zacina na prvnim radku metody a obepina celou tu danou metodu. 

vede to na dve veci: 

- bud ze vytvoris nejakou pomocnou fci ve ktere bude ten try catch
- lepsi error handler - i tady si dokazu predstavit ze budes mit jen jeden try catch a v catch jen koukas na to co za chybku se stalo

Reply:
upraveno